rotkehlchen/exchanges/binance.py

```diff
@@ -462,34 +474,30 @@ def api_query_list(
 
         return result
 
-    def _query_spot_balances(
+    def _add_balances(
             self,
             balances: defaultdict[AssetWithOracles, Balance],
+            new_balances: list[dict],
     ) -> defaultdict[AssetWithOracles, Balance]:
-        account_data = self.api_query_dict('api', 'account')
-        binance_balances = account_data.get('balances', None)
-        if not binance_balances:
-            raise RemoteError('Binance spot balances response did not contain the balances key')
-
-        for entry in binance_balances:
+        """Add new balances to balances dict"""
+        for entry in new_balances:
             try:
                 # force string https://github.com/rotki/rotki/issues/2342
                 asset_symbol = str(entry['asset'])
                 free = deserialize_asset_amount(entry['free'])
                 locked = deserialize_asset_amount(entry['locked'])
-            except KeyError as e:
-                raise RemoteError(f'Binance spot balance asset entry did not contain key {e!s}') from e  # noqa: E501
-            except DeserializationError as e:
-                raise RemoteError('Failed to deserialize an amount from binance spot balance asset entry') from e  # noqa: E501
+            except (KeyError, DeserializationError) as e:
+                raise RemoteError(
```

Comment:
Come on this is not gonna work. Please check how KeyError is handled in these cases in the code. Needs an if check.